src/overlay/FlowControlCapacity.cpp

```diff
@@ -0,0 +1,187 @@
+    uint64_t releasedFloodCapacity = 0;
+    size_t resourcesFreed = getMsgResourceCount(msg);
+    if (mCapacity.mTotalCapacity)
+    {
+        *mCapacity.mTotalCapacity += resourcesFreed;
+    }
+
+    if (mApp.getOverlayManager().isFloodMessage(msg))
+    {
+        if (mCapacity.mFloodCapacity == 0)
+        {
+            CLOG_DEBUG(Overlay, "Got flood capacity for peer {} ({})",
+                       mApp.getConfig().toShortString(mNodeID),
+                       mCapacity.mFloodCapacity + resourcesFreed);
+        }
+        releasedFloodCapacity = resourcesFreed;
+        mCapacity.mFloodCapacity += resourcesFreed;
+    }
+    checkCapacityInvariants();
+    return releasedFloodCapacity;
+}
+
+FlowControlCapacityMessages::FlowControlCapacityMessages(Application& app,
+                                                         NodeID nodeID)
+    : FlowControlCapacity(app, nodeID)
+{
+    mCapacity = getCapacityLimits();
+}
+
+FlowControlCapacity::ReadingCapacity
+FlowControlCapacityMessages::getCapacityLimits() const
+{
+    return {
+        mApp.getConfig().PEER_FLOOD_READING_CAPACITY,
+        std::make_optional<uint64_t>(mApp.getConfig().PEER_READING_CAPACITY)};
+}
+
+uint64_t
+FlowControlCapacityMessages::getMsgResourceCount(StellarMessage const& msg)
+{
+    // Each message takes one unit of capacity
+    return 1;
+}
+
+void
+FlowControlCapacityMessages::releaseOutboundCapacity(StellarMessage const& msg)
+{
+    ZoneScoped;
+    releaseAssert(msg.type() == SEND_MORE || msg.type() == SEND_MORE_EXTENDED);
+    auto numMessages = Peer::getNumMessages(msg);
+    if (!hasOutboundCapacity(msg) && numMessages != 0)
+    {
+        CLOG_DEBUG(Overlay, "Got outbound capacity for peer {}",
+                   mApp.getConfig().toShortString(mNodeID));
+    }
+    mOutboundCapacity += numMessages;
+}
+
+bool
+FlowControlCapacityMessages::hasOutboundCapacity(StellarMessage const& msg)
+{
+    return mOutboundCapacity > 0;
+}
+
+FlowControlCapacityBytes::FlowControlCapacityBytes(Application& app,
+                                                   NodeID nodeID)
+    : FlowControlCapacity(app, nodeID)
+{
+    mCapacity = getCapacityLimits();
+}
+
+FlowControlCapacity::ReadingCapacity
+FlowControlCapacityBytes::getCapacityLimits() const
+{
+    return {mApp.getConfig().PEER_FLOOD_READING_CAPACITY_BYTES, std::nullopt};
+}
+
+uint64_t
+FlowControlCapacityBytes::getMsgResourceCount(StellarMessage const& msg)
+{
+
+    return static_cast<uint64_t>(xdr::xdr_argpack_size(msg));
+}
+
+void
+FlowControlCapacityBytes::releaseOutboundCapacity(StellarMessage const& msg)
+{
+    ZoneScoped;
+    releaseAssert(msg.type() == SEND_MORE_EXTENDED);
+    if (!hasOutboundCapacity(msg) &&
+        (msg.sendMoreExtendedMessage().numBytes != 0))
+    {
+        CLOG_DEBUG(Overlay, "Got outbound capacity for peer {}",
+                   mApp.getConfig().toShortString(mNodeID));
+    }
+    mOutboundCapacity += msg.sendMoreExtendedMessage().numBytes;
```

Comment:
Can you add a few lines of comments (or a brief description in PR summary) regarding: sendMoreExtendedMessage + releaseOutbound() both in FlowControlCapacityBytes and FlowControlCapacity